pkg/api/vmtemplate/handler.go

```diff
@@ -0,0 +1,56 @@
+package vmtemplate
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/rancher/apiserver/pkg/handlers"
+	"github.com/rancher/apiserver/pkg/types"
+	ctlvmv1alpha1 "github.com/rancher/vm/pkg/generated/controllers/vm.cattle.io/v1alpha1"
+)
+
+type templateLinkHandler struct {
+	vmTemplateVersionCache ctlvmv1alpha1.VMTemplateVersionCache
+}
+
+func (h *templateLinkHandler) byIDHandler(request *types.APIRequest) (types.APIObject, error) {
+	vars := mux.Vars(request.Request)
+	if link, ok := vars["link"]; ok && link == "versions" {
+		if templateName, ok := vars["name"]; ok {
+			versions, err := h.getVersions(vars["namespace"], templateName)
```

Comment:
Can we get these vars from `request.Link/Namespace/Name`?

Reply:
updated